src/shared/Microsoft.Git.CredentialManager/Authentication/AuthenticationBase.cs

```diff
@@ -62,13 +62,28 @@ protected AuthenticationBase(ICommandContext context)
             return resultDict;
         }
 
+        protected void EnsureUserInteractionAllowed()
```

Comment:
This was not the pattern I expected, but now that I think about it more it makes sense. You're only going to show a user-prompt if you _need_ it to continue, but this setting prevents that from happening. The only rational thing to do is to exit with failure.

Reply:
Yes. GCM Core follows a `die` model when a bug or impossible situation (need to speak to the user, but cannot) is reached. Tracing contains the most detailed information.

I was thinking about renaming these methods to `ThrowIfUserInteractionDisabled()` to make their behaviour clearer? Thoughts?

Reply:
`ThrowIf` seems to be a clearer synonym of `Ensure`, so I'm happy with that rename.